connectors/sources/google_drive.py

```diff
@@ -557,100 +871,182 @@ async def get_content(self, blob, timestamp=None, doit=None):
+            else:
+                self._logger.warning(
+                    f"Unknown Google Drive permission type: {permission_type}."
+                )
+
+            processed_permissions.append(access_permission)
+
+        return processed_permissions
+
+    async def prepare_file(self, file, paths):
+        """Apply key mappings to the file document.
+
+        Args:
+            file (dict): File metadata returned from the Drive.
+
+        Returns:
+            dict: Formatted file metadata.
+        """
+
+        file_id = file.get("id")
+
+        file_document = {
+            "_id": file_id,
+            "created_at": file.get("createdTime"),
+            "last_updated": file.get("modifiedTime"),
+            "name": file.get("name"),
+            "size": file.get("size") or 0,  # handle folders and shortcuts
+            "_timestamp": file.get("modifiedTime"),
+            "mime_type": file.get("mimeType"),
+            "file_extension": file.get("fileExtension"),
+            "url": file.get("webViewLink"),
         }
 
         # record "file" or "folder" type
-        blob_document["type"] = (
-            "folder" if blob.get("mimeType") == FOLDER_MIME_TYPE else "file"
+        file_document["type"] = (
+            "folder" if file.get("mimeType") == FOLDER_MIME_TYPE else "file"
         )
 
         # populate owner-related fields if owner is present in the response from the Drive API
-        owners = blob.get("owners", None)
+        owners = file.get("owners", None)
         if owners:
-            first_owner = blob["owners"][0]
-            blob_document["author"] = ",".join(
+            first_owner = file["owners"][0]
+            file_document["author"] = ",".join(
                 [owner["displayName"] for owner in owners]
             )
-            blob_document["created_by"] = first_owner["displayName"]
-            blob_document["created_by_email"] = first_owner["emailAddress"]
+            file_document["created_by"] = first_owner["displayName"]
+            file_document["created_by_email"] = first_owner["emailAddress"]
 
         # handle last modifying user metadata
-        last_modifying_user = blob.get("lastModifyingUser", None)
+        last_modifying_user = file.get("lastModifyingUser", None)
         if last_modifying_user:
-            blob_document["updated_by"] = last_modifying_user.get("displayName", None)
-            blob_document["updated_by_email"] = last_modifying_user.get(
+            file_document["updated_by"] = last_modifying_user.get("displayName", None)
+            file_document["updated_by_email"] = last_modifying_user.get(
                 "emailAddress", None
             )
-            blob_document["updated_by_photo_url"] = last_modifying_user.get(
+            file_document["updated_by_photo_url"] = last_modifying_user.get(
                 "photoLink", None
             )
 
         # determine the path on google drive, note that google workspace files won't have a path
```

Comment:
So for a google workplace file on shared drive, `shared_drive` will be `None`, and there won't be any permissions field populated? 

If this is the case, can we just do something like:
```python
permissions = file.get("permissions", [])
if not permissions:
    permissions = await self._get_permissions_on_shared_drive(file_id=file_id)

file_document[ACCESS_CONTROL] = self._process_permissions(permissions)
```

Reply:
Yup, that seems to be the case. For files on shared drive the `file.get("permissions")` won't be populated. Sure I will simplify the logic